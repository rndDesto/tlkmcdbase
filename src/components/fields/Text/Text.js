import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Text(props) {
  const { className, input, inputProps, label, meta, helper } = props;
  const { dirty, error, touched } = meta;
  const classes =   [
    styles.root,
    !!input.value || styles.empty,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      <label>{label}</label>
      <input className={classes} id={input.name} {...input} {...inputProps}  />
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
      <span>{helper}</span>
    </div>
  );
}

Text.defaultProps = {
  className: '',
  helper:'',
  input: {},
  inputProps: {},
  label: '',
  meta:{},
};

Text.propTypes = {
  className: PropTypes.string,
  helper: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
};
