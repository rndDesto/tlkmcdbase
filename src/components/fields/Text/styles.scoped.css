.root {
  margin-bottom: 0.5rem;
}

.root > label {
  color: var(--palette-black-20);
  font-size: 0.875rem;
  line-height: 1rem;
  font-weight: bold;
}

.root > input {
  border: 0.0625rem solid var(--palette-text-light-2);
  border-radius: 0.25rem;
  box-shadow: none;
  color: var(--palette-black-10);
  display: block;
  font-size: 0.875rem;
  font-weight: bold;
  margin-top: 0.625rem;
  padding: 0.75rem 0.5rem;
  width: 100%;
  transition: border 0.25s;
}

.root > input[type="date"] {
  -webkit-appearance: none;
  background: url(/assets/icon-date.svg);
  background-position: calc(100% - 0.75rem);
  background-repeat: no-repeat;
  padding: 0.75rem;
  width: 100%;
}


input[type="date"]::-webkit-clear-button {
  display: none;
}

input[type="date"]::-webkit-inner-spin-button { 
  display: none;
}

input[type="date"]::-webkit-calendar-picker-indicator {
  width: 0;
  height: 0.2rem;
  position: relative;
  top: 0.1rem;
  right: 0.2rem;
  cursor: pointer;
}


.root > input:focus {
  border-color: var(--palette-base-primary-main);
}

.root > input::placeholder {
  color: var(--palette-text-light-2);
}

.root.error {
  border-color: var(--palette-textfield-error-main);
}

.root + small.error {
  color: var(--palette-textfield-error-main);
  display: block;
  font-size: 0.75rem;
  margin-bottom: 0.5rem;
}
