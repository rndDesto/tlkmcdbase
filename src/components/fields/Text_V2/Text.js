import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Text(props) {
  const { className, input, inputProps, label, meta, helper, icon } = props;
  const { dirty, error, touched } = meta;
  const classes =   [
    styles.root,
    !!input.value || styles.empty,
    !!error && (dirty || touched) && styles.error,
    className
  ].filter(Boolean).join(' ');
  return (
    <div className={classes}>
      <label>{label}</label>
      <div className={classes}>
        <input className={classes} id={input.name} {...input} {...inputProps}  />
        { icon ?
          <span>
            <img alt="icon" src={`../../../assets/${icon}.svg`}/>
          </span> : ''}
      </div>
      {!!error && (dirty || touched) && <small className={styles.error}>{error}</small>}
      <span>{helper}</span>
    </div>
  );
}

Text.defaultProps = {
  className: '',
  helper:'',
  icon:'',
  input: {},
  inputProps: {},
  label: '',
  meta:{},

};

Text.propTypes = {
  className: PropTypes.string,
  helper: PropTypes.string,
  icon: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,

};
