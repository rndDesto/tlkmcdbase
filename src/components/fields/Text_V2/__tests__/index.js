import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Text from '../Text';

describe('src/components/fields/Text', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Text />);
    expect(tree).toMatchSnapshot();

    const res1 = Text({
      ...Text.defaultProps,
      meta: { error: 'err', dirty: true, touched: true }
    });
    expect(res1.props.children[2].props.children).toBe('err');

    const res2 = Text({
      ...Text.defaultProps,
      meta: { error: 'err', dirty: false, touched: true }
    });
    expect(res2.props.children[2].props.children).toBe('err');
  });
});
