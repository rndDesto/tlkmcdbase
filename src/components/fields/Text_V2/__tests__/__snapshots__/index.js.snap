// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Text render 1`] = `
<div
  className=""
>
  <label>
    
  </label>
  <div
    className=""
  >
    <input
      className=""
    />
    
  </div>
  <span>
    
  </span>
</div>
`;
